ln-resource-mgr/src/forward_manager.rs

```diff
@@ -159,44 +159,146 @@ impl ForwardManagerImpl {
     ) -> Result<AllocationCheck, ReputationError> {
         forward.validate()?;
 
-        // Get the incoming revenue threshold that the outgoing channel must meet.
         let incoming_channel = self
             .channels
             .get_mut(&forward.incoming_ref.channel_id)
             .ok_or(ReputationError::ErrIncomingNotFound(
                 forward.incoming_ref.channel_id,
             ))?;
 
-        let revenue_threshold = incoming_channel
+        let incoming_revenue_threshold = incoming_channel
             .bidirectional_revenue
             .value_at_instant(forward.added_at)?;
 
+        // Reputation from the incoming channel.
+        let incoming_reputation = incoming_channel
+            .incoming_direction
+            .incoming_reputation(forward.added_at)?;
+
         let no_congestion_misuse = incoming_channel
             .incoming_direction
             .no_congestion_misuse(forward.added_at);
 
+        // Bucket resources available on the incoming channel.
+        let incoming_resources = ResourceValues {
```

Comment:
Can we cleanly move creation of `ResourceValues` into a helper function? 

I think it should be doable by passing in a `incoming_reputation: bool` and `&TrackedChannel`?

Reply:
Right, we only need `ReputationValues` for reputation in both directions 👍